Library.js

```diff
@@ -1,9 +1,126 @@
-let myLibrary = [];
+const myLibrary = [];
 
-function Book() {
-    // the constructor...
+function Book(title, author, numberPages, read, id) {
+  this.title = title;
+  this.author = author;
+  this.number_pages = numberPages;
+  this.read = read;
+  this.id = id;
 }
 
-function addBookToLibrary() {
-    // do stuff here
+Book.prototype.check_read = function checkRead() {
+  if (this.read === true) {
+    return 'already read';
+  }
+  return 'not read yet';
+};
+
+Book.prototype.opposite_status = function oppositeStatus() {
+  if (this.read === true) {
+    return 'Change to "Not read yet"';
+  }
+  return 'Change to "Read"';
+};
+
+
+Book.prototype.toggle_read = function toggleRead() {
+  this.read = !this.read;
+};
+
+Book.prototype.info = function info() {
+  return `${this.title} by ${this.author}, ${this.number_pages} pages, ${this.check_read(this.read)}.`;
+};
+
+const a = new Book('Harry Potter', 'IDK', 300, true, 0);
+const b = new Book('Rails', 'IDK', 300, true, 1);
+
+
+function addBookToLibrary(book) {
+  myLibrary.push(book);
+}
+
+function checkFirstId(element) {
+  if (element.length === 0) {
+    return 0;
+  }
+  return element[element.length - 1].id + 1;
+}
+
+function changeReadValue(value) {
+  if (value === 'true') {
+    return true;
+  }
+  return false;
+}
+
+function newBook() {
+  const title = document.getElementById('title').value;
+  const author = document.getElementById('author').value;
+  const pages = document.getElementById('pages').value;
+  let read = document.getElementById('read').value;
+  const id = checkFirstId(myLibrary);
+  read = changeReadValue(read);
```

Comment:
- [ ] **[OPTIONAL]** the form used to create a book should have validations, you should not be able to create an empty book. Let's add validate function for book before we add the book the the library.